h5py/tests/hl/test_vds/test_highlevel_vds.py

```diff
@@ -267,5 +267,59 @@ def test_slice_source(self):
     def tearDown(self):
         shutil.rmtree(self.tmpdir)
 
+@ut.skipUnless(vds_support,
+               'VDS requires HDF5 >= 1.9.233')
+class IndexingTestCase(ut.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp()
+        # Create source file (1.h5)
+        with h5.File(osp.join(self.tmpdir, '1.h5'), 'w') as f:
+            d = f.create_dataset('data', (10,), 'i4')
+            d[:] = np.arange(10)*10
+
+    def test_index_layout(self):
+        # Assemble virtual dataset (indexing target)
+        layout = h5.VirtualLayout((100,), 'i4')
+
+        inds = [3,6,20,25,33,47,70,75,96,98]
+
+        filename = osp.join(self.tmpdir, "1.h5")
+        vsource = h5.VirtualSource(filename, 'data', shape=(10,))
+        layout[inds] = vsource
+
+        outfile = osp.join(self.tmpdir, 'VDS.h5')
+
+        # Assembly virtual dataset (indexing source)
+        layout2 = h5.VirtualLayout((5,), 'i4')
+
+        inds2 = [0,1,4,5,8]
+        layout2 = vsource[inds2]
```

Comment:
This doesn't work, because you're replacing `layout2` instead of assigning into it. You need something like this:

```python
layout2[:] = vsource[inds2]
```